### PR TITLE
consolidates git provider options

### LINKED ISSUE #6693: Create a single overview page explaining git provider options to users
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/docs/cloud/git/import-a-project-by-git-url#section-managing-deploy-keys
https://docs.getdbt.com/docs/cloud/git/git-configuration-in-dbt-cloud

### What part(s) of the page would you like to see updated?

Feedback from product design (Michael Stein) on updating the configure git overview page:

We definitely need an update to the docs for repository connections.

This is specifically for the onboarding flow, so we need to have docs that are for connecting GitHub, GitLab, Managed, Git clone, and Azure DevOps. And it would be great if those were all in one doc, or at least centralized so we can use just one link.
https://docs.getdbt.com/docs/cloud/git/import-a-project-by-git-url#section-managing-deploy-keys
This is the current doc that we link to, but it's not very friendly for beginner users, and also doesn't mention Managed or Git clone and how those work.

we should definitely change it to link to that page. If we could add something about managed that would be great, and if it could say something about Git clone rather than "Import a project by git URL" I think that would ease some confusion for users, since that's how we talk about it in the product.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr consolidates git provider options to one page. previously the configure git page didn't include managed repo option. 

it also updates the page from 'import via git url' to 'git clone, which matches the dbt cloud ui. 

redirects added and links changed too.

requested by  by Michael Stein

Resolves #6693

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/docs/cloud/dbt-cloud-ide/ide-user-interface
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/docs/cloud/git/git-configuration-in-dbt-cloud
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/docs/cloud/git/import-a-project-by-git-url
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/docs/cloud/git/managed-repository
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/docs/collaborate/git/managed-repository
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/docs/collaborate/git/pr-template
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/faqs/Git/gitignore
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/faqs/Troubleshooting/ide-session-unknown-error
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/guides/dbt-python-snowpark
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/guides/sl-snowflake-qs
- https://docs-getdbt-com-git-update-git-setup-dbt-labs.vercel.app/guides/snowflake-qs

### CODE REVIEW

File: website/docs/docs/cloud/git/git-configuration-in-dbt-cloud.md [hunk 1]
```diff
@@ -1,37 +1,46 @@
...
     title="Connect to GitLab"
-    body="Learn how to connect to GitLab."
+    body="Learn how to connect to GitLab using dbt Cloud's native integration."
     link="/docs/cloud/git/connect-gitlab"
     icon="dbt-bit"/>
 
 <Card
-    title="Connect to Azure DevOps"
-    body="Learn how to connect to Azure DevOps."
+    title="Connect to Azure DevOps "
```
Comment (mirnawong1) [inline website/docs/docs/cloud/git/git-configuration-in-dbt-cloud.md]:
```suggestion
    title="Connect to Azure DevOps"
```

File: website/docs/faqs/Git/gitignore.md [hunk 2]
```diff
@@ -51,7 +53,9 @@ For more info on `gitignore` syntax, refer to the [Git docs](https://git-scm.com
 
 <Lightbox src="/img/docs/dbt-cloud/cloud-ide/gitignore-italics.png" width="50%" title="A dbt project on the main branch that has properly configured gitignore folders (highlighted in italics)."/>
 
-### Fix in the git provider
+</Expandable>
+
+<Expandable alt_header="Fix in the git provider">
```
Comment (matthewshaver) [inline website/docs/faqs/Git/gitignore.md]: I realize it's lower case elsewhere but I'll grab this one now and the rest post-holiday
```suggestion
<Expandable alt_header="Fix in the Git provider">
```

File: website/docs/docs/cloud/git/managed-repository.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+id: "managed-repository"
+description: "Learn how to set up a project with a managed repository."
+pagination_next: "docs/cloud/git/import-a-project-by-git-url"
+pagination_prev: "docs/cloud/git/git-configuration-in-dbt-cloud"
+---
+
+If you don't already have a Git repository for your dbt project, you can let dbt Cloud manage a repository for you. Managed repositories are a great way to trial dbt without needing to create a new repository.
+
+:::info
+We do not recommend using a managed repository in a production environment. You will not be able to use git features like pull requests which are part of our recommended version control best practices.
```
Comment (matthewshaver) [inline website/docs/docs/cloud/git/managed-repository.md]:
```suggestion
dbt Labs recommends against using a managed repository in a production environment. You can't use Git features like pull requests, which are part of our recommended version control best practices.
```

File: website/docs/faqs/Git/gitignore.md (hunk 2 above)
Comment (mirnawong1) [reply website/docs/faqs/Git/gitignore.md]: good shout!

File: website/docs/docs/cloud/git/managed-repository.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+:::
+
+To set up a project with a managed repository:
+
+1. From your A**ccount settings** in dbt Cloud, select the project you want to set up with a managed repository. If the project already has a repository set up, you need to edit the repository settings and disconnect the existing repository.
```
Comment (matthewshaver) [inline website/docs/docs/cloud/git/managed-repository.md]:
```suggestion
1. From your **Account settings** in dbt Cloud, select the project you want to set up with a managed repository. If the project already has a repository set up, you need to edit the repository settings and disconnect the existing repository.
```

Comment (mirnawong1) [reply website/docs/docs/cloud/git/managed-repository.md]: yikes good catch!

File: website/docs/docs/cloud/git/managed-repository.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+1. From your **Account settings** in dbt Cloud, select the project you want to set up with a managed repository. If the project already has a repository set up, you need to edit the repository settings and disconnect the existing repository.
+2. Click **Edit** for the project.
+3. Under Repository, click **Configure repository**.
+4. Select **Managed**.
+5. Enter a name for the repository. For example, "analytics" or "dbt-models."
+6. Click **Create**.
+   <Lightbox src="/img/docs/dbt-cloud/cloud-configuring-dbt-cloud/managed-repo.png" title="Adding a managed repository"/>
+
+- dbt Cloud will host and manage this repository for you.
```
Comment (matthewshaver) [inline website/docs/docs/cloud/git/managed-repository.md]: Removing some future tense
```suggestion
- dbt Cloud hosts and manages this repository for you.
```

File: website/docs/docs/cloud/git/git-configuration-in-dbt-cloud.md [hunk 6]
```diff
@@ -1,37 +1,46 @@
...
+title: "Configure Git in dbt Cloud"
 description: "Learn about the Git providers supported in dbt Cloud"
-pagination_next: "docs/cloud/git/import-a-project-by-git-url"
-pagination_prev: null
+hide_table_of_contents: true
+pagination_next: "docs/cloud/git/managed-repository"
 ---
 
-<div className="grid--2-col">
+[Version control](/docs/collaborate/git/version-control-basics) is an essential part of the dbt workflow, enabling teams to collaborate effectively and maintain a history of changes to their dbt projects.
```
Comment (matthewshaver) [inline website/docs/docs/cloud/git/git-configuration-in-dbt-cloud.md]: Not a blocker here, but in the future,e we might want to open with a little blurb about what Git is

**Review by matthewshaver — APPROVED** [review]

File: website/docs/docs/cloud/git/git-configuration-in-dbt-cloud.md [hunk 7]
```diff
@@ -1,37 +1,46 @@
...
+title: "Configure Git in dbt Cloud"
 description: "Learn about the Git providers supported in dbt Cloud"
-pagination_next: "docs/cloud/git/import-a-project-by-git-url"
-pagination_prev: null
+hide_table_of_contents: true
+pagination_next: "docs/cloud/git/managed-repository"
 ---
 
-<div className="grid--2-col">
+[Version control](/docs/collaborate/git/version-control-basics) &mdash; a system allows you and your teammates to work collaboratively, safely, and simultaneously on a single project &mdash; is an essential part of the dbt workflow, enabling teams to collaborate effectively and maintain a history of changes to their dbt projects.
```
Comment (mirnawong1) [inline website/docs/docs/cloud/git/git-configuration-in-dbt-cloud.md]:
```suggestion
[Version control](/docs/collaborate/git/version-control-basics) &mdash; a system allows you and your teammates to work safely and simultaneously on a single project &mdash; is an essential part of the dbt workflow. It enables teams to collaborate effectively and maintain a history of changes to their dbt projects. 
```

File: website/docs/docs/cloud/git/git-configuration-in-dbt-cloud.md [hunk 8]
```diff
@@ -1,37 +1,46 @@
...
+title: "Configure Git in dbt Cloud"
 description: "Learn about the Git providers supported in dbt Cloud"
-pagination_next: "docs/cloud/git/import-a-project-by-git-url"
-pagination_prev: null
+hide_table_of_contents: true
+pagination_next: "docs/cloud/git/managed-repository"
 ---
 
-<div className="grid--2-col">
+[Version control](/docs/collaborate/git/version-control-basics) &mdash; a system allows you and your teammates to work safely and simultaneously on a single project &mdash; is an essential part of the dbt workflow. It enables teams to collaborate effectively and maintain a history of changes to their dbt projects.
```
Comment (mirnawong1) [inline website/docs/docs/cloud/git/git-configuration-in-dbt-cloud.md]:
```suggestion
[Version control](/docs/collaborate/git/version-control-basics) &mdash; a system that allows you and your teammates to work safely and simultaneously on a single project &mdash; is an essential part of the dbt workflow. It enables teams to collaborate effectively and maintain a history of changes to their dbt projects. 
```

**Pull request merged**